packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -217,17 +231,20 @@ function useFocusContainment(scopeRef: RefObject<HTMLElement[]>, contain: boolea
     };
 
     let onFocus = (e) => {
-      // If a focus event occurs outside the active scope (e.g. user tabs from browser location bar),
-      // restore focus to the previously focused node or the first tabbable element in the active scope.
-      let isInAnyScope = isElementInAnyScope(e.target, scopes);
-      if (!isInAnyScope) {
+      // If focusing an element in a child scope of the currently active scope, the child becomes active.
+      // Moving out of the active scope to an ancestor is not allowed.
+      if (!activeScope || isAncestorScope(activeScope, scopeRef)) {
+        activeScope = scopeRef;
+        focusedNode.current = e.target;
+      } else if (scopeRef === activeScope && !isElementInScope(e.target, scopeRef.current)) {
+        // If a focus event occurs outside the active scope (e.g. user tabs from browser location bar),
+        // restore focus to the previously focused node or the first tabbable element in the active scope.
```

Comment:
Did some digging on my previous [comment](https://github.com/adobe/react-spectrum/pull/2223#pullrequestreview-733198966). Looks like the nested menu is actually opening but then immediately closing. What seems to be happening is that `useFocusContainment`'s `scopeRef` doesn't have the Menu's newly added scope and thus judges the menu item focus event to be outside the active scope and not within a child scope of the currently active scope. This means the menu trigger button gets refocused, automatically closing the menu

Reply:
Ugh, I had this story working earlier. Will check again